### PR TITLE
[fix](default value) change throw AnalysisException order

### LINKED ISSUE #35758: [Bug] report unintuitive errors when types other than DATE and DATEV2 use current_date as the default value during creating a table.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master branch: 78431dbb80d9db00c411297c951c63b9b11d01d7

### What's Wrong?

Types other than DATE and DATEV2 use current_date as the default value when create a table. Some literal constructor will throw AnalysisException. It is not intuitive to users.

### What You Expected?

throw an AnalysisException similar to the following:
```sql
Types other than DATE and DATEV2 cannot use current_date as the default value
```

### How to Reproduce?

```sql
mysql> create table t1(a int, b double default current_date);
ERROR 1105 (HY000): errCode = 2, detailMessage = errCode = 2, detailMessage = Invalid floating-point literal: CURRENT_DATE
```

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Dragonliu2018**
## Proposed changes

Issue Number: close #35758 

when types other than DATE and DATEV2 use current_date as the default value during creating a table, throw an AnalysisException similar to the following:
```sql
ERROR 1105 (HY000): errCode = 2, detailMessage = errCode = 2, detailMessage = Types other than DATE and DATEV2 cannot use current_date as the default value
```
but not this AnalysisException:
```sql
ERROR 1105 (HY000): errCode = 2, detailMessage = errCode = 2, detailMessage = Invalid floating-point literal: CURRENT_DATE

```

### CODE REVIEW

**Comment by Dragonliu2018** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java [hunk 1]
```diff
@@ -480,8 +480,30 @@ public static void validateDefaultValue(Type type, String defaultValue, DefaultV
         ScalarType scalarType = (ScalarType) type;
 
         // check if default value is valid.
-        // if not, some literal constructor will throw AnalysisException
+        // first, check if the type of defaultValue matches primitiveType.
+        // if not check it first, some literal constructor will throw AnalysisException,
+        // and it is not intuitive to users.
         PrimitiveType primitiveType = scalarType.getPrimitiveType();
+        if (null != defaultValueExprDef && defaultValueExprDef.getExprName().equals("now")) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java:487]:
```suggestion
        if (null != defaultValueExprDef && defaultValueExprDef.getExprName().equalsIgnoreCase("now")) {
```

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java [hunk 2]
```diff
@@ -480,8 +480,30 @@ public static void validateDefaultValue(Type type, String defaultValue, DefaultV
...
+            switch (primitiveType) {
+                case DATETIME:
+                case DATETIMEV2:
+                    break;
+                default:
+                    throw new AnalysisException("Types other than DATETIME and DATETIMEV2 "
+                            + "cannot use current_timestamp as the default value");
+            }
+        } else if (null != defaultValueExprDef
+                && defaultValueExprDef.getExprName().equals(DefaultValue.CURRENT_DATE.toLowerCase())) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java:497]:
```suggestion
                && defaultValueExprDef.getExprName(). equalsIgnoreCase(DefaultValue.CURRENT_DATE)) {
```

**Comment by Dragonliu2018** [comment]
run buildall

**Comment by morrySnow** [comment]
add test case please

**Review by morrySnow — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**